src/fkooman/VootProvider/LdapVootStorage.php

```diff
@@ -167,8 +191,19 @@ public function isMemberOf($resourceOwnerId, $startIndex = null, $count = null)
         $userDn = $this->getUserDn($resourceOwnerId);
 
         $userGroups = array();
+        
+        $groupsProvider = $this->config->s('LdapVootStorage')->l('groupsProvider');
+    
         /* get the groups the user is a member of */
-        $filter = '(' . $this->config->s('LdapVootStorage')->l('memberAttribute') . '=' . $userDn . ')';
+        switch ($groupsProvider):
+            case "posixgroup":
+                $filter = '(' . $this->config->s('LdapVootStorage')->l('memberAttribute') . '=' . $resourceOwnerId . ')';
+            break;
+            default:
+                $filter = '(' . $this->config->s('LdapVootStorage')->l('memberAttribute') . '=' . $userDn . ')';
+            break;
+        endswitch;
```

Comment:
Can you use `{` and `}` to define blocks to be consistent with the other code?

I'm wondering if `switch` really adds value here or that just using if/else would suffice.